src/bin/addr2line.rs

```diff
@@ -61,11 +57,16 @@ fn main() {
     let exe = matches.value_of("executable").unwrap_or("./a.out");
     let show_funcs = matches.is_present("functions");
 
-    let debug = if show_funcs {
-        addr2line::Mapping::with_functions(path::Path::new(&exe))
-    } else {
-        addr2line::Mapping::new(path::Path::new(&exe))
-    };
+    let mut opts = addr2line::Options::default();
+
+    if show_funcs {
```

Comment:
Just for consistency, I think I'd prefer this to be `matches.is_present("functions")`, and then to do the `let show_funcs` along with `let show_addrs` below (i.e., right above the `let one =`). It does mean that we repeat `is_present("functions")`, but the code reads nicer.